src/main.py

```diff
@@ -1,26 +1,49 @@
 # Standard Packages
-import sys
-import json
+import sys, json, yaml
 from typing import Optional
 
 # External Packages
 import uvicorn
-from fastapi import FastAPI
+from fastapi import FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.staticfiles import StaticFiles
+from fastapi.templating import Jinja2Templates
 
 # Internal Packages
 from src.search_type import asymmetric, symmetric_ledger, image_search
 from src.utils.helpers import get_absolute_path
 from src.utils.cli import cli
 from src.utils.config import SearchType, SearchModels, TextSearchConfig, ImageSearchConfig, SearchConfig, ProcessorConfig, ConversationProcessorConfig
+from src.utils.rawconfig import FullConfig
 from src.processor.conversation.gpt import converse, message_to_log, message_to_prompt, understand
 
-
 # Application Global State
 model = SearchModels()
 search_config = SearchConfig()
```

Comment:
Maybe we can collate all these `config` related global variables into a single global variable/class/object? Don't have to do it as part of this PR necessarily, just a note for later reference

Reply:
Opened #13.